R/sort_levels.R

```diff
@@ -0,0 +1,67 @@
+#' Sort factor levels according to a user-defined function
+#'
+#' @param .f A factor (or character vector).
+#' @param .fun A function that will sort or permute the existing factor levels.
+#'   It must accept one character argument and return a character argument of
+#'   the same length as it's input.
+#'
+#' @param ... Additional arguments to `.fun`.
+#' @export
+#' @examples
+#' medieval_experiment <- factor(
+#'   c("Control", "Comfrey", "Cropleek", "Bishopwort")
+#' )
+#' medieval_experiment
+#'
+#' # Default (alphabetical) level-sorting:
+#' fct_sort_levels(medieval_experiment, sort)
+#'
+#' # Reversed ordering (equivalent to `fct_rev`):
```

Comment:
It's not in general equivalent to `fct_rev()`, so I think it'd be better to just remove that comparison